web/src/screens/private/Groups/EditGroup.js

```diff
@@ -3,39 +3,328 @@ import PropTypes from 'prop-types';
+
+    let tags = [];
+    let usersToAdd = [];
+
+    if (regions !== '') {
+      tags = tags.concat(regions.split(',').map(tag => ({ id: parseInt(tag, 10) })));
+    }
+
+    if (capabilities !== '') {
+      tags = tags.concat(capabilities.split(',').map(tag => ({ id: parseInt(tag, 10) })));
+    }
+
+    if (users !== '') {
+      usersToAdd = users.split(',').map(user => ({ id: parseInt(user, 10) }));
+    }
+
+    if (this.state.id === 0) {
+      this.props
+        .createGroup({
+          name,
+          tags,
+          users: usersToAdd,
+        })
+        .then(() => {
+          const { history } = this.props;
+          history.push('/groups');
+        })
+        .catch((error) => {
+          this.setState(() => ({ message: error.message, open: true }));
+          setTimeout(() => {
+            this.setState(() => ({ message: '', open: false }));
+          }, 3000);
+        });
+    } else {
+      this.props
+        .updateGroup({
+          id: this.state.id,
+          name,
+          tags,
+          users: usersToAdd,
+        })
+        .then(() => {
+          const { history } = this.props;
+          history.push('/groups');
+        })
+        .catch((error) => {
+          this.setState(() => ({ message: error.message, open: true }));
+          setTimeout(() => {
+            this.setState(() => ({ message: '', open: false }));
+          }, 3000);
+        });
+    }
+  };
+
+  render() {
+    const { name } = this.state;
+    const { classes, loading, user } = this.props;
+
+    if (loading) {
+      return <CircularProgress className={classes.progress} size={50} />;
+    }
+
+    const capabilities = user.organisation.tags
+      .filter(tag => tag.type === TAG_TYPE_CAPABILITY)
+      .map(tag => ({ value: tag.id.toString(), label: tag.name }));
+
+    const regions = user.organisation.tags
+      .filter(tag => tag.type === TAG_TYPE_ORG_STRUCTURE)
+      .map(tag => ({ value: tag.id.toString(), label: tag.name }));
+
+    const users = user.organisation.users.map(u => ({
+      value: u.id.toString(),
+      label: u.displayName,
+    }));
+
+    return (
+      <div className={classes.root}>
+        <div className={classes.actionPanel}>
+          <Typography variant="title">{this.state.id === 0 ? 'Add New' : 'Edit'} Group</Typography>
+        </div>
+        <Paper className={classes.paperForm}>
+          <FormControl className={classes.formControl}>
+            <TextField
+              id="name"
+              className={classes.textField}
+              value={name}
+              onChange={e => this.handleChange(e, 'name')}
+              margin="normal"
+              placeholder="Enter Group..."
+            />
+          </FormControl>
+          <FormControl className={classes.formControl}>
+            <Tag
+              list={regions}
+              label="Regions"
+              placeholder="Select..."
+              onChange={this.handleTagChange('regions')}
```

Comment:
dont call function in render

Reply:
e.g. I did similar things in SignUpPage.js etc a long time ago 


            <TextField
              required
              fullWidth
              id="password"
              label="Password"
              type="password"
              margin="normal"
              value={this.state.password}
              onChange={this.onPasswordChange}
            />

Reply:
nevermind - worked out the issue in the react docs